core/templates/services/guppy-configuration.service.spec.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit test for GuppyConfigurationService
+ */
+
+import { TestBed } from '@angular/core/testing';
+
+import { GuppyConfigurationService } from
+  'services/guppy-configuration.service';
+
+declare global {
+  interface Window {
+    Guppy: Object;
+  }
+}
+
+describe('GuppyConfigurationService', () => {
+  let guppyConfigurationService: GuppyConfigurationService = null;
+  class MockGuppy {
+    static 'remove_global_symbol'(symbol: string): void {}
+  }
+
+  beforeEach(() => {
+    window.Guppy = MockGuppy;
+    spyOn(Guppy, 'remove_global_symbol');
+    guppyConfigurationService = TestBed.get(GuppyConfigurationService);
+  });
+
+  it('should configure guppy if service is not initialized', () => {
+    GuppyConfigurationService.serviceIsInitialized = false;
+    guppyConfigurationService.init();
+    expect(Guppy.remove_global_symbol).toHaveBeenCalled();
+  });
+
+  it('should not configure guppy if service is initialized', () => {
```

Comment:
Add a test that actually tries to check the scenario we are interested in, i.e. we call init() multiple times from different components.

(In general, if you don't write enough tests, your reviewer will just ask you to add more tests, so I would advise you to think through and proactively cover as many cases as you can from the outset. When in doubt, add more.)

Reply:
Done.